txpipe/photoz_stack.py

```diff
@@ -60,18 +62,32 @@ def run(self):
             self.config['chunk_rows']  # number of rows to read at once
         )
 
-        tomography_iterator = self.iterate_hdf(
-            'tomography_catalog', # tag of input file to iterate through
+        photoz_iterator2 = self.iterate_hdf(
```

Comment:
it looks like you're looping through all the data twice here, which is not needed.  Loading the entire data set is quite slow, so we try to minimize it.  You can combine these in the same loop.

Reply:
ahhh, ok I think I understand that now, changed